### ISSUE TITLE
Refactor / cleanup implementation

### ISSUE THREAD

**Issue by stefanbesler** [stefanbesler/struckig#10]
- Do a better job at separating user api and internals. At the moment almost everything is publicly accessible.
- Check for byval parameters
- Remove temporary variables in favor of instance variables (only after #9 to have a comparison)
- Unify naming conventions such that merging from upstream is easy, but doesn’t clash too much with (my) structured-text conventions
- Remove static classes with member variables (e.g. Roots) in favour of actual static classes without members of traditional function blocks. The code has to be thread-safe.

**Comment by stefanbesler** [comment]
Already done some time ago
